### DISCUSSION TITLE
TLS connection works well, but some clients resulted  Unexpected Message\n {unsupported_record_type,71}"}}}

### DISCUSSION THREAD

**Discussion by yukuochang** [emqx/emqx#10400] [Q&A]
emqx version 4.4.14.

as title described, somthing wrong happend, logs as following:
2023-04-14T08:06:00.015652+00:00 [error] supervisor: 'esockd_connection_sup - <0.3298.0>', errorContext: connection_shutdown, reason: {ssl_error,{tls_alert,{unexpected_message,"TLS server: In state hello at tls_record.erl:564 generated SERVER ALERT: Fatal - Unexpected Message\n {unsupported_record_type,71}"}}}, offender: [{pid,<0.24330.1>},{name,connection},{mfargs,{emqx_connection,start_link,[[{deflate_options,[]},{crl_options,[{crl_check_enabled,true},{crl_cache_urls,["http://iidp-provider-credential-service:8864/api/v1/certificate/crl/clients.crl"]}]},{ocsp_options,[{ocsp_stapling_enabled,false},{ocsp_refresh_interval,300000},{ocsp_refresh_http_timeout,15000}]},{max_conn_rate,500},{active_n,100},{zone,external},{proxy_address_header,<<>>},{proxy_port_header,<<>>},{supported_subprotocols,[]},{peer_cert_as_username,cn},{peer_cert_as_clientid,cn}]]}}]

the test client  issued as following, what happened?
![9C84F173307938191151B1CCD04501D7](https://user-images.githubusercontent.com/55518766/231999924-14d2705b-530c-49e4-bae3-43cf132bcf46.png)

**Comment by ieQu1** [comment]
Hello,

```
SERVER ALERT: Fatal - Unexpected Message\n {unsupported_record_type,71}"}}}
```
This error message usually appears when plaintext MQTT traffic is sent to TLS-enabled port.
However, in your screenshot it looks like you send TLS-encrypted traffic to TLS port, which is correct.
Questions:
1) Is 61.135.163.10 IP of EMQX broker, or a load balancer of some sort?
2) Do `{unsupported_record_type` errors correlate with your attempts to run mosquitto, in other words, is this error caused by this command, or perhaps some other unrelated client, like port scanner or web crawler? It's pretty common that publicly facing MQTT brokers get requests from search engine crawlers.

**Reply by yukuochang** [reply]
i stopped all clients, but stiil get connections at night, so it's maybe caused by port scanner or web crawler

**Comment by yukuochang** [comment]
thank you for your reply!
1. 61.135.163.10 is a vip supplied by a gateway called BGW, which is a load balancing device implemented with software. BGW virtualizes multiple servers (i.e. RealServers) into one server, providing a unified VIP (i.e. virtual IP). Users only need to communicate with the VIP to access the services on the RealServer.  61.135.163.10 can be accessed normally through common physical computers, but the problem occurd when accessed  though a cloud host.
2.  I stopped all client connections to test this issue. but at night ,when it's not used ,message like this still be produced , so, the error message can'not be confirmed to have been triggered by this command.

**Comment by savonarola** [comment]
Hello!
Sorry, do I understand the presented scenario correctly:
* On `instance-gp8sng8z` host we try to make an **SSH connection** to 61.135.163.10:8883
* We fail (`banner line contains invalid characters`) and stay on `instance-gp8sng8z` host
* Then we try to make an MQTT connection with mosquito  to **the same** host:port 61.135.163.10:8883


If 61.135.163.10:8883 has EMQX behind, could the error message (`"TLS server: In state hello at tls_record.erl:564 generated SERVER ALERT: Fatal - Unexpected Message\n {unsupported_record_type,71}"`) be triggered with these unsuccessful SSH connections to the MQTT tls port?

**Reply by yukuochang** [reply]
the SSH command is only used to confirm that  the endpoint (61.135.163.10:8883)  is available for instance-gp8sng8z, and it's only issued for one time.
    but i got lots of error in the log  which looks the same as that of "Unexpected Message\n {unsupported_record_type,71}", so oboviously  it's  related to  using of mosquitto_pub for many times

**Reply by yukuochang** [reply]
maybe caused by port scanner or web crawler

**Comment by yukuochang** [comment]
Another test for raw package captured from network interface card with 10.233.108.105 as real ip that behind VIP 61.135.163.10:
 
normal connection from physical computer：
![c2b4c099b41cb83a077f6f0f7a5d5118](https://user-images.githubusercontent.com/55518766/232691795-7e6f155f-1167-4ea7-bd68-909d27eaee79.jpg)

failed connection from cloud host：
<img width="1384" alt="60780686ef8853a60ac75c1fde77066b" src="https://user-images.githubusercontent.com/55518766/232692026-b381ebe3-dc34-4b79-8662-640330755081.png">
![image](https://user-images.githubusercontent.com/55518766/232704274-82a61b2b-3815-4228-9b19-ef1005b2907f.png)

![image](https://user-images.githubusercontent.com/55518766/232706205-c7eafb31-8226-40ab-924e-d7c346b1e2b6.png)

![image](https://user-images.githubusercontent.com/55518766/232711413-83ccc857-d955-4c1c-a29c-01db813c929e.png)

![image](https://user-images.githubusercontent.com/55518766/232710229-76debf01-f28c-45d7-8525-30cc61ff7b5f.png)

![image](https://user-images.githubusercontent.com/55518766/232709107-8993abcf-318d-49a4-9584-5eb337ac5084.png)
![image](https://user-images.githubusercontent.com/55518766/232709406-2d625f69-3271-4f57-a813-99014b7427ad.png)

it seems the TLS version can only be version TLS1.3 and both server's cert and client's cert are version TLS1.3.
but the cloud host does not have Cipher suite for TLSv1.3,  maybe the openssl version should be greater than v1.1.1

**Comment by yukuochang** [comment]
Another cloud host with version expeced:
![7982F4CE2CB9312F97602E1BF8CF55FA](https://user-images.githubusercontent.com/55518766/232746235-c23a5edd-7868-4400-b584-792ed863bb02.png)
![E864D40803214E63A5AB3BD8023BC3C6](https://user-images.githubusercontent.com/55518766/232746269-e1186fbf-6872-442d-af79-2fedfa5404a7.png)
![image](https://user-images.githubusercontent.com/55518766/232746994-79d90993-6348-4dfa-875b-0ec001746811.png)

**Comment by yukuochang** [comment]
solved，just a network problem, that caused lost of tcp fragment
